src/ardrive.ts

```diff
@@ -164,38 +179,196 @@ export class ArDrive extends ArDriveAnonymous {
 			],
 			tips: [tipData],
 			fees: {
-				[uploadFileResult.metaDataTrxId]: +uploadFileResult.metaDataTrxReward,
 				[uploadFileResult.dataTrxId]: +uploadFileResult.dataTrxReward,
+				[uploadFileResult.metaDataTrxId]: +uploadFileResult.metaDataTrxReward,
 				[tipData.txId]: +communityTipTrxReward
 			}
 		});
 	}
 
+	public async createPublicFolderAndUploadChildren(
+		parentFolderId: FolderID,
+		wrappedFolder: FsFolder,
+		parentFolderName?: string
+	): Promise<ArFSResult> {
+		const results = await this.recursivelyCreatePublicFolderAndUploadChildren({
+			parentFolderId,
+			wrappedFolder,
+			parentFolderName
+		});
+
+		const { tipData, reward: communityTipTrxReward } = await this.sendCommunityTip(results.communityWinstonTip);
+
+		return Promise.resolve({
+			created: results.entityResults,
+			tips: [tipData],
+			fees: { ...results.feeResults, [tipData.txId]: +communityTipTrxReward }
+		});
+	}
+
+	protected async recursivelyCreatePublicFolderAndUploadChildren({
+		parentFolderId,
+		wrappedFolder,
+		parentFolderName,
+		driveId
+	}: CreatePublicFolderAndUploadChildrenParams): Promise<{
+		entityResults: ArFSEntityData[];
+		feeResults: ArFSFees;
+		communityWinstonTip: Winston;
+	}> {
+		// DriveID will not exist if this folder is the parent folder
+		// Recursing children folders will have driveId assigned by the parent folder
+		const isParentFolder: boolean = driveId === undefined;
+
+		if (!driveId) {
+			// Retrieve drive ID from folder ID and ensure that it is indeed public
+			driveId = await this.arFsDao.getDriveIdForFolderId(parentFolderId);
+			const drive = await this.arFsDao.getPublicDrive(driveId);
+			if (!drive) {
+				throw new Error(`Public drive with Drive ID ${driveId} not found!`);
+			}
+		}
+
+		const parentFolderData = new ArFSPublicFolderTransactionData(
+			parentFolderName ?? wrappedFolder.getBaseFileName()
+		);
+
+		let communityWinstonTip = 0;
+
+		if (isParentFolder) {
```

Comment:
Lift this into createPublicFolderAndUploadChildren now that it exists. This will make it easier to confirm the price later, and avoid passing the tip around needlessly